src/main/scala/dpla/ingestion3/mappers/providers/JsonRetconMapper.scala

```diff
@@ -0,0 +1,210 @@
+package dpla.ingestion3.mappers.providers
+
+import dpla.ingestion3.mappers.utils.{Document, JsonExtractor, JsonMapping}
+import dpla.ingestion3.model.DplaMapData.{AtLeastOne, ExactlyOne, ZeroToMany, ZeroToOne}
+import dpla.ingestion3.model.{DcmiTypeCollection, DplaPlace, EdmAgent, EdmTimeSpan, EdmWebResource, LiteralOrUri, SkosConcept, URI, eitherStringOrUri, nameOnlyAgent, nameOnlyConcept, nameOnlyPlace, uriOnlyWebResource}
+import org.json4s.JValue
+import org.json4s.JsonAST.{JArray, JObject, JString}
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods.{compact, render}
+
+abstract class JsonRetconMapper extends JsonMapping with JsonExtractor {
+
+  // OreAggregation
+  override def dplaUri(data: Document[JValue]): ZeroToOne[URI] = {
+    data.get \ "_source" \ "@id" match {
+      case JString(value) => Some(URI(f"http://dp.la/api/items/$value"))
+      case _ => None
+    }
+  }
+
+  override def sidecar(data: Document[JValue]): JValue =
+    ("prehashId", buildProviderBaseId()(data)) ~
+      ("dplaId", extractString(data.get \ "_source" \ "id").getOrElse(""))
```

Comment:
There should be no `.getOrElse` on this -- we want the extraction on the jsonlRecord() method to fail if the value is not provided and empty string will validate.